pkg/cloudprovider/aws/aws.go

```diff
@@ -429,6 +436,17 @@ func (aws *AWSCloud) ExternalID(name string) (string, error) {
 	return *inst.InstanceID, nil
 }
 
+// InstanceID returns the cloud provider ID of the specified instance.
+func (aws *AWSCloud) InstanceID(name string) (string, error) {
+	inst, err := aws.getInstancesByDnsName(name)
+	if err != nil {
+		return "", err
+	}
+	// In the future it is possible to also return an endpoint as:
+	// <endpoint>/<zone>/<instanceid>
+	return "/" + *inst.Placement.AvailabilityZone + "/" + *inst.InstanceID, nil
```

Comment:
There's a comment above: in the future the format may be: `<endpoint>/<zone>/<instanceid>`
Therefore now for aws we would have: `aws:///<zone>/<instanceid>` because `<endpoint>` is empty.